### PR TITLE
Initial build/setup files for worker pool

### PR DESCRIPTION
**Pull request by rranjan3**
- Add docker-compose-pool.yaml which runs single KME &
  WPE in worker pool
- Introduce ENCLAVE_TYPE for build time decision making
  for different enclave types
- Replicate and customize setup files for WPE/KME/SWE

Signed-off-by: Rajeev Ranjan <rajeev2.ranjan@intel.com>

### CODE REVIEW

File: docker-compose-pool.yaml [hunk 1]
```diff
@@ -0,0 +1,151 @@
+# Copyright 2019 Intel Corporation
```
Comment (danintel) [inline docker-compose-pool.yaml:1]: s/2019/2019-2020/

File: enclave_manager/setup_kme.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python
+
+# Copyright 2018 Intel Corporation
```
Comment (danintel) [inline enclave_manager/setup_kme.py:3]: s/2018/2018-2020/

File: listener/listener_config.toml [hunk 3]
```diff
@@ -29,6 +29,8 @@ max_work_order_count = 10
 # ZMQ configurations the listener would connect to
 # Same as the url and port of enclave manager socket
 zmq_url = "tcp://avalon-enclave-manager:"
+# url to use for worker pool when deploying using docker-compose-pool.yaml
```
Comment (danintel) [inline listener/listener_config.toml:32]: Lines 32 and 33 should be before line 31.

File: docker-compose-pool.yaml [hunk 4]
```diff
@@ -0,0 +1,151 @@
+# Copyright 2019-2020 Intel Corporation
```
Comment (manojgop) [inline docker-compose-pool.yaml:1]: Need to follow the approach in https://github.com/hyperledger/avalon/commit/1ba5385f33fe510be942d1675392659fb1e4bb99

File: enclave_manager/setup_kme.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    undef_macros = ['NDEBUG', 'EDEBUG']
+    )
+
+## -----------------------------------------------------------------
+version = subprocess.check_output(
+    os.path.join(tcf_root_dir, 'bin/get_version')).decode('ascii').strip()
+
+setup(name='kme_enclave_manager',
+      version = version,
+      description = 'Avalon Intel SGX Enclave Manager',
```
Comment (manojgop) [inline enclave_manager/setup_kme.py:110]: Enclave Manager -> KME Enclave Manager

File: enclave_manager/setup_singleton.py [hunk 6]
```diff
@@ -109,7 +105,7 @@
 version = subprocess.check_output(
     os.path.join(tcf_root_dir, 'bin/get_version')).decode('ascii').strip()
 
-setup(name='avalon_enclave_manager',
+setup(name='singleton_enclave_manager',
       version = version,
       description = 'Avalon Intel SGX Enclave Manager',
```
Comment (manojgop) [inline enclave_manager/setup_singleton.py:110]: Enclave Manager -> Pinned/Singleton Enclave Manager

File: enclave_manager/setup_wpe.py [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+    undef_macros = ['NDEBUG', 'EDEBUG']
+    )
+
+## -----------------------------------------------------------------
+version = subprocess.check_output(
+    os.path.join(tcf_root_dir, 'bin/get_version')).decode('ascii').strip()
+
+setup(name='wpe_enclave_manager',
+      version = version,
+      description = 'Avalon Intel SGX Enclave Manager',
```
Comment (manojgop) [inline enclave_manager/setup_wpe.py:110]: Enclave Manager -> WPE Enclave Manager

File: docker-compose.yaml [hunk 8]
```diff
@@ -53,6 +53,7 @@ services:
         - http_proxy
         - https_proxy
         - no_proxy
+        - ENCLAVE_TYPE=singleton
```
Comment (manojgop) [inline docker-compose.yaml:56]: Create docker-compose-singleton.yaml and follow the approach in https://github.com/hyperledger/avalon/commit/1ba5385f33fe510be942d1675392659fb1e4bb99

**Pull request merged**